From fa7da180ff87bcbe4454af0de583a61ed9247778 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 8 Jul 2026 16:58:32 +0200
Subject: [PATCH 34/46] AnnaThermostat: add
 intended_domestic_hot_water_comfort_mode measurement

Map intended_domestic_hot_water_comfort_mode to a Domoticz switch device.

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
Assisted-by: GPT-5.3-Codex

Upstream-Status: Inappropriate [WIP]
---
 hardware/AnnaThermostat.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/hardware/AnnaThermostat.cpp b/hardware/AnnaThermostat.cpp
index 8aa8e81c1..dd8800ee7 100644
--- a/hardware/AnnaThermostat.cpp
+++ b/hardware/AnnaThermostat.cpp
@@ -1214,6 +1214,15 @@ void CAnnaThermostat::GetMeterDetails()
 					SendSwitch(applianceIDhex, 6, batterypercentage, isOn, 0, ApplianceName + " " + sname, m_Name);
 				}
 			}
+			else if (sname == "intended_domestic_hot_water_comfort_mode")
+			{
+				if (!tmpstr.empty())
+				{
+					bool isOn = false;
+					if (isEnabledState(tmpstr, isOn))
+						SendSwitch(applianceIDhex, 13, batterypercentage, isOn, 0, ApplianceName + " " + sname, m_Name);
+				}
+			}
 			else if (sname == "domestic_hot_water_state")
 			{
 				if (!tmpstr.empty())
